lib/Service/EncryptService.php

```diff
@@ -206,13 +206,24 @@ protected function getKeys($salt, $key) {
 		return array($cipherKey, $macKey, $iv);
 	}
 
+    /**
+     * Use Double HMAC Comparison with a random key to truly blind the comparison operation.
+     * It is not strictly required by using hash_equals (https://www.php.net/manual/en/function.hash-equals.php),
+     * but it is a second layer of security to prevent timing attacks.
+     *
+     * @param string $a
+     * @param string $b
+     *
+     * @return bool
+     * @throws \Exception
+     */
 	protected function hash_equals($a, $b) {
```

Comment:
I believe this funtion was an implementation with compatibility with older php versions, it wouldn't be needed nowadays since we don't support that versions in the codebase anyways.

The proper way to change it would be to check if the hash_equals function exist and if so call that and if not, use the old implementation.

But as I said, i believe this was a way to do what hash_equals does but on older php versions.

Reply:
so you don't agree with Double HMAC Comparison?

I mean actually I can remove that and we use php's hash_equals() method instead. Easy way.
Not sure if the Double HMAC Comparison is really that mega profitable xD

thought it would be a cool idea